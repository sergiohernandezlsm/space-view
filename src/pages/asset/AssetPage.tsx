import { useContext, useEffect } from "react";
import AssetsContext from "../../store/assets-context";
import { useParams } from "react-router-dom";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Card from "../../components/card";
import AudioCard from "../../components/audioCard";
import useGetAssetMetaData from "../../hook/useGetAssetMetaData";
import styles from "./AssetPage.module.scss";

export interface ParamTypes {
  id: string;
}

const AssetPage = () => {
  const { metaData, getMetaData } = useGetAssetMetaData();
  const ctx = useContext(AssetsContext);
  const { id } = useParams<ParamTypes>();

  useEffect(() => {
    if (ctx.assetData.mediaType) {
      getMetaData(id, ctx.assetData.mediaType);
    }
  }, [ctx.assetData.mediaType, getMetaData, id]);

  const card =
    ctx.assetData.mediaType === "image" ? (
      <Card imageLocation={ctx.assetData.href} />
    ) : (
      <AudioCard href={ctx.assetData.href} />
    );

  return (
    <Container className={styles.assetPageWrapper}>
      <Row>
        <h1>{metaData.title}</h1>
        <p>{metaData.description}</p>
        {card}
      </Row>
    </Container>
  );
};

export default AssetPage;
