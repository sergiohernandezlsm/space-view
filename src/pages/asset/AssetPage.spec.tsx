import { useEffect } from "react";
import { shallow } from "enzyme";
import AssetPage from "./AssetPage";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    id: "test",
  }),
}));

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
    assetData: {
      href: "test",
      mediaType: "audio",
    },
  }),
  useEffect: jest.fn(),
}));

describe("AssetPage", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useEffect as jest.Mock).mockImplementation((f) => f());
  });
  it("should render AssetPage", () => {
    const wrapper = shallow(<AssetPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
