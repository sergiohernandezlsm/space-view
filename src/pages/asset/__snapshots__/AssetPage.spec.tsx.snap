// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetPage should render AssetPage 1`] = `ShallowWrapper {}`;
